crates/hyperdrive-math/src/short/max.rs

```diff
@@ -115,21 +116,41 @@ impl State {
             open_vault_share_price,
             maybe_conservative_price,
         );
-        for _ in 0..maybe_max_iterations.unwrap_or(7) {
+        let mut previous_max_bond_amount = max_bond_amount;
+        let mut step_size = fixed!(1e18);
+        for _ in 0..maybe_max_iterations.unwrap_or(10) {
             let deposit = match self.calculate_open_short(
                 max_bond_amount,
                 spot_price,
                 open_vault_share_price,
             ) {
                 Ok(d) => d,
-                Err(_) => return max_bond_amount,
+                Err(_) => {
+                    // The pool is insolvent at this point
+                    // we break out of the function for final checks
+                    // and return
+                    break;
+                }
             };
-            max_bond_amount += (budget - deposit)
-                / self.short_deposit_derivative(
-                    max_bond_amount,
-                    spot_price,
-                    open_vault_share_price,
-                );
+
+            // If we overshot, we undo a step and cut the
+            // step size by half
```

Comment:
This is fine IMO, but it might be better to just proceed with Newton's method and properly handle crossing over. We'd just do something like this:

```
if budget >= deposit {
                max_bond_amount += (budget - deposit)
                / self.short_deposit_derivative(
                    max_bond_amount,
                    spot_price,
                    open_vault_share_price,
                );
} else {
                    max_bond_amount -= (deposit - budget)
                / self.short_deposit_derivative(
                    max_bond_amount,
                    spot_price,
                    open_vault_share_price,
                );
}
```

The derivative might need to be updated in this case, but you could just switch to `I256`. It's up to y'all, but Newton's method is going to have better convergence properties.

Reply:
You can see what we do here: https://github.com/delvtech/hyperdrive/blob/ed2bc6d4387d6de00670869ccea8a548873bacd5/contracts/src/libraries/LPMath.sol#L862.

Reply:
The issue with this is that only one side is valid, so we really only want to approach from one side only (the side where budget >= deposit). Recovering back to a valid state when on the other side is more difficult.

Reply:
We can just record the best value on the right side. It'll look similar to what we do in `LPMath`. Just because we only want to take values less than the max doesn't mean we can't leverage Newton's convergence. It should start flipping back and forth.

It's just a suggestion.

Reply:
Implemented the suggestion above, with an added bias to the budget to avoid the case where newton's method never crosses back over to the valid side.